### PR TITLE
fix: add subjextAltName to webhook.crt

### LINKED ISSUE #15: Fix certificate for K8S 1.19
Issue when deploying to Kuberntes 1.19:

`x509: certificate relies on legacy Common Name field, use SANs`

This is due to Kubernetes updating to GoLang 1.15: https://kubernetes.io/docs/setup/release/notes/#api-change

```txt
Kubernetes is now built with golang 1.15.0-rc.1.
The deprecated, legacy behavior of treating the CommonName field on X.509 serving certificates as a host name when no Subject Alternative Names are present is now disabled by default. It can be temporarily re-enabled by adding the value x509ignoreCN=0 to the GODEBUG environment variable. (#93264, @justaugustus) [SIG API Machinery, Auth, CLI, Cloud Provider, Cluster Lifecycle, Instrumentation, Network, Node, Release, Scalability, Storage and Testing]
```

Kubesec-webhook is still stable in prior versions of Kubernetes.

### PR DESCRIPTION
**Pull request by denhamparry**
Closes #15 

- webhook failing in k8s 1.19
- issue with GoLang 1.15: x509: certificate relies on legacy Common Name field, use SANs
- set subjectAltName in x509 cert
- tested in latest releases of k8s 1.16 to 1.19
- kubectl --dry-run is deprecated and can be replaced with --dry-run=client

### CODE REVIEW

File: deploy/gen-certs.sh [hunk 1]
```diff
@@ -7,15 +7,23 @@ OS="`uname`"
...
+  -out ./webhookCA.csr
+openssl x509 -req \
+  -extfile <(printf "subjectAltName=DNS:%s.%s.svc" "${NAME}" "${NAMESPACE}") \
+  -days 365 \
+  -in webhookCA.csr \
+  -signkey webhookCA.key \
+  -out webhook.crt
 
 # Generate cert secret
 kubectl -n kubesec create secret generic \
```
Comment (ipedrazas) [inline deploy/gen-certs.sh:22]: If the command depends on the target version of Kubernetes, the script should have an option/switch to manage it accordingly.

**Review by ipedrazas — CHANGES_REQUESTED** [review]
The script should check that the conditions to safely run are met. Check k8s versions, if not right, fail. Added a comment in the code.

File: deploy/gen-certs.sh (hunk 1 above)
Comment (denhamparry) [reply deploy/gen-certs.sh:30]: Will test the PR against previous versions of Kubernetes to test that the certificates work with previous versions.

Comment (denhamparry) [reply deploy/gen-certs.sh:32]: The test worked on versions 1.16 - 1.20 of Kubernetes.

**Comment by denhamparry** [comment]
Closing PR, recreating new PR due to changing requirements

**Pull request closed without merging**